### PR TITLE
docs(en): add Binary Search

### PR DESCRIPTION
**Pull request by UG-SEP**
**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] New algorithm
- [ ] Optimization in previous algorithms
- [ ] Code style update
- [ ] Refactor
- [x] Documentation
- [ ] Other, please describe:

**Briefly describe the changes in this PR**

Created a Searching folder in docs/en/ and added Binary Search doc in it

**Other information:**

closes: #289

### CODE REVIEW

File: docs/en/Searching/Binary-Search.MD [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+Middle element:**30** and element is less then 30 so search between start to middle -1 index
+
+Middle element: **20** and yes the middle element is the key to found so return the index=**1**
+
+Output: **1**
+
+## Implementation
+
+[Binary Search in C](https://github.com/MakeContributions/DSA/blob/main/algorithms/C/searching/Binary-search.c)
```
Comment (ming-tsai) [inline docs/en/Searching/Binary-Search.MD:43]: Could you just put the language? as below example:
```suggestion
[C](https://github.com/MakeContributions/DSA/blob/main/algorithms/C/searching/Binary-search.c)
```

Comment (ming-tsai) [reply docs/en/Searching/Binary-Search.MD:43]: And convert it to a list

File: docs/en/Searching/Binary-Search.MD [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+- [Go](https://github.com/MakeContributions/DSA/blob/main/algorithms/Go/searching/binary-search.go)
+
+- [Java](https://github.com/MakeContributions/DSA/blob/main/algorithms/Java/searching/binary-search.java)
+
+- [JavaScript](https://github.com/MakeContributions/DSA/blob/main/algorithms/JavaScript/src/searching/binary-search.js)
+
+- [JavaScript](https://github.com/MakeContributions/DSA/blob/main/algorithms/JavaScript/src/searching/binary-search-recursive.js)
+
+- [Python](https://github.com/MakeContributions/DSA/blob/main/algorithms/Python/searching/binary_search.py)
```
Comment (ming-tsai) [inline docs/en/Searching/Binary-Search.MD:57]: Could you remove the new line in the middle of each option?
```suggestion
- [C](https://github.com/MakeContributions/DSA/blob/main/algorithms/C/searching/Binary-search.c)
- [C++](https://github.com/MakeContributions/DSA/blob/main/algorithms/CPlusPlus/Searching/binary-search.cpp)
- [CSharp](https://github.com/MakeContributions/DSA/blob/main/algorithms/CSharp/src/Search/binary-search.cs)
- [Go](https://github.com/MakeContributions/DSA/blob/main/algorithms/Go/searching/binary-search.go)
- [Java](https://github.com/MakeContributions/DSA/blob/main/algorithms/Java/searching/binary-search.java)
- [JavaScript](https://github.com/MakeContributions/DSA/blob/main/algorithms/JavaScript/src/searching/binary-search.js)
- [JavaScript](https://github.com/MakeContributions/DSA/blob/main/algorithms/JavaScript/src/searching/binary-search-recursive.js)
- [Python](https://github.com/MakeContributions/DSA/blob/main/algorithms/Python/searching/binary_search.py)
```

File: docs/en/Searching/Binary-Search.MD [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+2. Check whether the key is equal to middle element if yes then return the index and exit the program
+
+3. If the 2 step didn't run then test whether the element is less than the middle element if yes then run the step: 1 between the start to middle-1 index 
+
+4. If the 3 step didn't run then test whether the element is high than the middle element if yes then run the step: 1 between the middle+1 to last index.
+
+5. Run the loop till the starting index is less than end index
+
+6. If the loop over and data not found then return -1 that means data doesn't exist
```
Comment (ming-tsai) [inline docs/en/Searching/Binary-Search.MD:23]: Could you remove the new line in the middle of each option?

```suggestion
1. Find the middle element of the array
2. Check whether the key is equal to the middle element if yes then return the index and exit the program
3. If the 2 steps didn't run then test whether the element is less than the middle element if yes then run the step: 1 between the start to middle-1 index 
4. If the 3 steps didn't run then test whether the element is high than the middle element if yes then run the step: 1 between the middle+1 to the last index.
5. Run the loop till the starting index is less than the end index
6. If the loop over and data not found then return -1 that means data doesn't exist
```

File: docs/en/Searching/Binary-Search.MD [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+   **Note:** The array should be sorted in ascending to descending order
```
Comment (ming-tsai) [inline docs/en/Searching/Binary-Search.MD:25]: You could use `>`
```suggestion
> **Note:** The array should be sorted in ascending to descending order
```

**Review by ming-tsai — APPROVED** [review]
Nice work @UG-SEP

**Comment by UG-SEP** [comment]
Thanks @ming-tsai 
Really I want to thanks all member they all help me a lot at this is the project where I started learning how to contribute in a nice way and @Arsenic-ATG is one of the best reviewers I saw till 
He is really awesome 
Thanks one again to all

**Pull request merged**